p2p/service/gossip_listener.go

```diff
@@ -0,0 +1,74 @@
+package service
+
+import (
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/priorityq"
+	"sync"
+)
+
+// GossipDataHandler is the function type that will be called when data is
+type GossipDataHandler func(data GossipMessage, syncer Syncer)
+
+// Listener represents the main struct that reqisters delegates to gossip function
+type Listener struct {
+	*log.Log
+	net      Service
+	channels []chan GossipMessage
+	stoppers []chan struct{}
+	syncer   Syncer
+	wg       sync.WaitGroup
+}
+
+// NewListener creates a new listener struct
+func NewListener(net Service, syncer Syncer, log log.Log) *Listener {
+	return &Listener{
+		Log:    &log,
+		net:    net,
+		syncer: syncer,
+		wg:     sync.WaitGroup{},
+	}
+}
+
+// Syncer is interface for sync services
+type Syncer interface {
+	FetchAtxReferences(atx *types.ActivationTx) error
+	FetchPoetProof(poetProofRef []byte) error
+	ListenToGossip() bool
+	IsSynced() bool
```

Comment:
I only see this this method being used from this interface.. maybe we can cut some of it ?

Reply:
all of them are used by different handlers